go/ai/option.go

```diff
@@ -430,6 +446,20 @@ func WithOutputFormat(format OutputFormat) OutputOption {
 	return &outputOptions{OutputFormat: format}
 }
 
+// WithOutputInstructions sets which, if any, output instructions should be added.
+// Nil/not set: default instructions will be added.
+// Empty string: no instructions will be added.
+// Non-empty string: will be used as instructions.
+func WithOutputInstructions(instructions *string) OutputOption {
+	return &outputOptions{OutputInstructions: instructions}
+}
+
+// WithOutputNativeConstrained sets whether to use model native constrained output generation.
+// If this is set to true and the model does not support native constrained generation, it will be simulated instead. Defaults to true.
+func WithOutputNativeConstrained(constrained bool) OutputOption {
```

Comment:
I wonder if this should be something like `WithUseSimulatedConstrained()` with no arg. Since native is the default and we still fallback to simulated if the model chosen doesn't support native, there's no much point to passing in a bool.